main.go

```diff
@@ -247,7 +248,8 @@ func compile(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, sketc
 
 	out := svg
 	if filepath.Ext(outputPath) == ".png" {
-		svg := svg
+		svg := appendix.AppendTooltips(diagram, ruler, svg)
```

Comment:
Why not call `d2svg` with an extra argument like `withTooltipAppendix` instead of manipulating raw HTML/SVG?

Reply:
more composable. d2svg.Render would get real messy with all the flags.

Reply:
I won't enter a long discussion on the composition as it seems quite hackish to split the html and append back, so I'll assume this is ok for now and we can refactor later on if we need something else

Reply:
yeah but that's two separate things. like it should not be an XML string until the very end. d2svg.Render should return a structured XML that postprocessing steps can manipulate.

Reply:
it makes sense, though at the same time the interface could be actual composition/chain of functions where you don't need to parse, use regexes and change viewports after it was ready.
In this case, `main` would decide which functions to include in the pipeline
```
sag := Render([diagram, tooltips])

func diagram(buffer) {
    buffer.write(...)
}

func tooltips(buffer) {
    buffer.write(...)
}
```

Reply:
> parse, use regexes and change viewports

it would still need to do all that in the proposed pseudocode

Reply:
Why? if the buffer is always appending to the end.
`Render` opens `<svg>` and closes `</svg>` everything else happens in between

Reply:
oh i see. yeah that'd be a good refactor to do.